task-sdk/src/airflow/sdk/exceptions.py

```diff
@@ -18,15 +18,32 @@
 from __future__ import annotations
 
 import enum
+from http import HTTPStatus
 from typing import TYPE_CHECKING, Any
 
-from airflow.exceptions import AirflowException
 from airflow.sdk import TriggerRule
 
 if TYPE_CHECKING:
+    from collections.abc import Collection
+
+    from airflow.sdk.definitions.asset import AssetNameRef, AssetUniqueKey, AssetUriRef
     from airflow.sdk.execution_time.comms import ErrorResponse
 
 
+class AirflowException(Exception):
```

Comment:
I wonder if this would be easier for future compat if we made this? Dunno

```suggestion
class AirflowException(RuntimeError):
```